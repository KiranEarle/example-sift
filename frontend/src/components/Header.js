import React from 'react'
import PropTypes from 'prop-types'

const Header = ({counts}) => {
  return (
    <div className='header'>
      <div className='header-stats'>
      <h2>Message Total</h2>
        <div>{counts.messageTotal}</div>
      </div>
      <div className='header-stats'>
      <h2>Word Total</h2>
        <div>{counts.wordTotal}</div>
      </div>
      <div className='header-stats'>
      <h2>Total</h2>
        <div>{counts.wpmTotal}</div>
      </div>
    </div>
  )
}

Header.PropTypes = {
  /**
   * Takes in an object containing details of the messages
   */
  counts: PropTypes.object
}

export default Header