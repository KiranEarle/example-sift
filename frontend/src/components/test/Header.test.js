import React from 'react'
import Header from '../Header'

it('render with props without blowing up', () => {

  const props = {
    messageTotal: 23,
    wordTotal: 343,
    wpmTotal: 433
  }

  const component = shallow(<Header counts={props} />);
  expect(component).toMatchSnapshot();
});