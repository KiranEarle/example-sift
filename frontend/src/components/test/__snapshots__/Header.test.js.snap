// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with props without blowing up 1`] = `
<div
  className="header"
>
  <div
    className="header-stats"
  >
    <h2>
      Message Total
    </h2>
    <div>
      23
    </div>
  </div>
  <div
    className="header-stats"
  >
    <h2>
      Word Total
    </h2>
    <div>
      343
    </div>
  </div>
  <div
    className="header-stats"
  >
    <h2>
      Words/Message
    </h2>
    <div>
      433
    </div>
  </div>
</div>
`;
